vispy/visuals/instanced_mesh.py

```diff
@@ -0,0 +1,157 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+
+"""An instanced version of MeshVisual with arbitrary shifts, transforms, and colors."""
+
+from __future__ import division
+
+import numpy as np
+
+from ..gloo import VertexBuffer
+from ..gloo.texture import downcast_to_32bit_if_needed
+from ..color import ColorArray
+from .filters import InstancedShadingFilter
+
+from .mesh import MeshVisual
+
+
+_VERTEX_SHADER = """
+uniform bool use_instance_colors;
+
+// these attributes will be defined on an instance basis
+attribute vec3 shift;
+attribute vec4 instance_color;
+attribute vec3 transform_x;
+attribute vec3 transform_y;
+attribute vec3 transform_z;
+
+vec4 color;
+
+varying vec4 v_base_color;
+void main() {
+    if (use_instance_colors) {
+        color = instance_color;
+    } else {
+        color = $base_color;
+    }
+
+    v_base_color = $color_transform(color);
+
+    // transform is generated from column vectors (new basis vectors)
+    // https://en.wikibooks.org/wiki/GLSL_Programming/Vector_and_Matrix_Operations#Constructors
+    mat3 instance_transform = mat3(transform_x, transform_y, transform_z);
+    vec3 pos_rotated = instance_transform * $to_vec4($position).xyz;
+    vec4 pos_shifted = $to_vec4(pos_rotated + shift);
+    gl_Position = $transform(pos_shifted);
+}
+"""
+
+
+class InstancedMeshVisual(MeshVisual):
+    """Instanced Mesh visual.
+
+    Mostly identical to MeshVisual, but additionally takes arrays of
+    of positions and transforms (optionally colors) to create multiple
+    instances of the mesh.
+
+    Parameters
+    ----------
+    instance_positions : (I, 3) array
+        Coordinates for each instance of the mesh.
+    instance_transforms : (I, 3, 3) array
+        Matrices for the transforms to apply to each instance.
+    instance_colors : ColorArray
+        Matrices of colors for each instance. Colors
+    *args : list
+        Positional arguments to pass to :class:`MeshVisual`.
+    **kwargs : dict
+        Keyword arguments to pass to :class:`MeshVisual`.
```

Comment:
I checked the CI rendered version and it didn't seem to like this. Let's try:

```suggestion
    *args : list
        Positional arguments to pass to :class:`vispy.visuals.MeshVisual`.
    **kwargs : dict
        Keyword arguments to pass to :class:`vispy.visuals.MeshVisual`.
```